### PR TITLE
Auth Backend: Refactor google provider

### LINKED ISSUE #1054: Auth Backend: Refactor Google provider to reduce dependency on passport
PassportJS is a black box when it comes to using it as a middleware and handling errors so refactor to get more control over error flow and reduce dependency on passportJS.

- use passport only for the provider strategy
- remove the need for middlewares in auth-backend/service/router.ts

### PR DESCRIPTION
**Pull request by soapraj**
- Refactor google auth to reduce passport dependency.
- Deleting existing tests for now as this is almost a complete rewrite. 
- new tests will follow in a separate PR once we settle on the implementation

#### :heavy_check_mark: Checklist

- [ ] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [ ] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

File: plugins/auth-backend/src/providers/OAuthProvider.ts [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+  const options: CookieOptions = {
+    maxAge: TEN_MINUTES_MS,
+    secure: false,
+    sameSite: 'none',
+    domain: 'localhost',
+    path: `/auth/${provider}/handler`,
+    httpOnly: true,
+  };
+
+  res.cookie(`${provider}-nonce`, nonce, options);
```
Comment (freben) [inline plugins/auth-backend/src/providers/OAuthProvider.ts:51]: These cookies should have the environment id  as part of their name right? So you can have both ghe and github.com auths in flight at the same time and with the same underlying provider

File: plugins/auth-backend/src/providers/types.ts [hunk 2]
```diff
@@ -22,40 +22,26 @@ export type AuthProviderConfig = {
...
 };
 
-export interface AuthProvider {
-  strategy(): passport.Strategy;
-  router?(): express.Router;
+export interface OAuthProviderHandlers {
+  start(req: express.Request, options: any): Promise<any>;
+  handler(req: express.Request): Promise<any>;
+  refresh(refreshToken: string, scope: string): Promise<any>;
+  logout?(): Promise<any>;
```
Comment (freben) [inline plugins/auth-backend/src/providers/types.ts:29]: Doc comments everywhere, same for the route ones before

**Review by freben — APPROVED** [review]
Looks good. The execution stuff is a little magic, but that's an effect of laying bare the express internals I guess

File: plugins/auth-backend/src/providers/OAuthProvider.ts (hunk 1 above)
Comment (Rugvip) [reply plugins/auth-backend/src/providers/OAuthProvider.ts:51]: Yeah the provider ID here should end up being some integrator-provided value. So provider id could be ghe or github, but the provider type would be github for both, for example.

**Pull request merged**